File: src/primes.go
package main

import (
	"fmt" // Println
	"os"  // Args
	"strconv" // Atoi
	"math" // Sqrt
)

func main() {

	max, _ := strconv.Atoi(os.Args[1])

	nums := make([]bool, max)

	// 0 and 1 are not primes
	nums[0] = true
	nums[1] = true

	// Computing primes
	for i := 2; i < int(math.Sqrt(float64(max)))+1; i++ {
		if nums[i] { continue }
		for j := i*2; j < max; j += i {
			nums[j] = true
		}
	}

	// Counting prime numbers
	total := 0
	for i := 0; i < max; i++ {
		if !nums[i] {
			total++
		}
	}
	fmt.Println(total)
}

File: src/primes.rs
use std::env;

fn main() {

    let max : usize = env::args().nth(1).unwrap().parse().unwrap();

    let mut nums = vec![false; max];

    // 0 and 1 are not primes
    nums[0] = true;
    nums[1] = true;

    // Computing prime numbers
    for i in 2..(1+(max as f64).sqrt() as usize) {
        if nums[i] { continue; }
        for j in (i*2..max).step_by(i) {
            nums[j] = true;
        }
    }

    // Couting the prime numbers
    let mut total: u32 = 0;
    for i in 0..max {
        if !nums[i] {
            total += 1;
        }
    }
    print!("{}",total);
}
